### ISSUE TITLE
[Model Enhancement]: Chocolate Classification using DL

### ISSUE THREAD

**Issue by Arihant-Bhandari** [abhisheks008/DL-Simplified#553]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : [Model Enhancement]: Chocolate Classification using DL</br>
:red_circle: **Aim** : enhancing custom CNN model</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/siddharthmandgi/chocolate-classification</br>
:red_circle: **Approach** : adding attention as overhead to CNN architecture and using ELU and NADAM in activation and optimizer.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : Arihant Bhandari
- GitHub Profile Link : Arihant-Bhandari https://github.com/Arihant-Bhandari
- Email ID : arihant.bhandari.2004@gmail.com
- Participant ID (if applicable): Arihant Bhandari (eternal_insight)
- Approach for this Project : Creating a CNN-Attention model for bettering CNN accuracy
- What is your participant role? (Mention the Open Source program)
GSSoC 2024 Contributor
*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by Arihant-Bhandari** [comment]
@abhisheks008 hi, can i have a go at this, i feel attention can help boost accuracy for CNN. If i find my work substantial and satisfactory i will PR, if not i will close the issue on my end.

thank you for your time.

**Comment by abhisheks008** [comment]
Are you planning to enhance this project, https://github.com/abhisheks008/DL-Simplified/tree/main/Chocolate%20Classification%20using%20DL right?

**Comment by Arihant-Bhandari** [comment]
yes @abhisheks008 , the CNN is at 0.86 currently, i think it can be boosted to 0.94 ish with attention, if i do end up achieving this i will start PR.

**Comment by abhisheks008** [comment]
Follow the same representation like you did in the previous issue. Go ahead with this. Assigned to you @Arihant-Bhandari

**Comment by Arihant-Bhandari** [comment]
@abhisheks008 hi, i think i have completed the project, it has been boosted as i expected, but i want to send over the file without the actual PR for you to see before we make things official with PR.

is it possible, like with an email or on discord ?

**Comment by abhisheks008** [comment]
> @abhisheks008 hi, i think i have completed the project, it has been boosted as i expected, but i want to send over the file without the actual PR for you to see before we make things official with PR.
> 
> is it possible, like with an email or on discord ?

Chill dude, make a PR, if any changes are required you can update them flawlessly.

**Comment by Arihant-Bhandari** [comment]
i had 2 possible solutions actually both are same implementation its just model structuring differences. how about i sent them over , you can have a look at them @abhisheks008.

**Comment by abhisheks008** [comment]
You can sent them over abhishek.opensource@gmail.com. My suggestion is put both of them together in the `.ipynb` file, why not showcase a comparison. That'll be insightful.

**Comment by Arihant-Bhandari** [comment]
cool we can do this, i will push them and send a PR in, thanks for listening @abhisheks008

**Comment by Arihant-Bhandari** [comment]
@abhisheks008 i have sent the files over email, pls check, also i noticed someone was working on the README file, issue #526, should i wait for them before updating readme with metrics like i did in the previous project or do it on my own ? i can hand them metrics and image link if they are working on it.

**Comment by abhisheks008** [comment]
> @abhisheks008 i have sent the files over email, pls check, also i noticed someone was working on the README file, issue #526, should i wait for them before updating readme with metrics like i did in the previous project or do it on my own ? i can hand them metrics and image link if they are working on it.



> @abhisheks008 i have sent the files over email, pls check, also i noticed someone was working on the README file, issue #526, should i wait for them before updating readme with metrics like i did in the previous project or do it on my own ? i can hand them metrics and image link if they are working on it.



> @abhisheks008 i have sent the files over email, pls check, also i noticed someone was working on the README file, issue #526, should i wait for them before updating readme with metrics like i did in the previous project or do it on my own ? i can hand them metrics and image link if they are working on it.

Let's wait for that issue to solved.

**Comment by Arihant-Bhandari** [comment]
hi @abhisheks008 , have you reviewed the codebase, which should i be pushing as PR ?

**Comment by abhisheks008** [comment]
I checked the code and found that you have overfitted the train dataset as a result it is giving an accuracy of 100% which is an eye catching event. Try to increase the test dataset portion.

**Comment by Arihant-Bhandari** [comment]
hi @abhisheks008 im starting my PR i think i am on the right track here, pls review the codebase, if any changes needed i will be more than glad to do things over. thank you for your patience.
